src/tools/rustbook/src/main.rs

```diff
@@ -46,23 +50,39 @@ fn main() {
 
                 ::std::process::exit(101);
             }
-        },
+        }
         ("linkcheck", Some(sub_matches)) => {
             if let Err(err) = linkcheck(sub_matches) {
                 eprintln!("Error: {}", err);
 
-                #[cfg(feature = "linkcheck")]
-                {
-                    if let Ok(broken_links) = err.downcast::<BrokenLinks>() {
-                        for cause in broken_links.links().iter() {
-                            eprintln!("\tCaused By: {}", cause);
-                        }
+                // HACK: ignore timeouts
+                let actually_broken = {
+                    #[cfg(feature = "linkcheck")]
+                    {
+                        err.downcast::<BrokenLinks>()
+                            .map(|broken_links| {
+                                broken_links
+                                    .links()
+                                    .iter()
+                                    .inspect(|cause| eprintln!("\tCaused By: {}", cause))
+                                    .any(|cause| !format!("{}", cause).contains("timed out"))
+                            })
+                            .unwrap_or(false)
                     }
-                }
 
-                ::std::process::exit(101);
+                    #[cfg(not(feature = "linkcheck"))]
+                    {
+                        false
```

Comment:
Can this be `unreachable!`, just to be clear?

Reply:
It's not unreachable, or am I wrong?. The idea is that actually_broken is set to false when not using the linkcheck feature.

Reply:
The `linkcheck()` function never returns `Err`, so this code block shouldn't ever be entered, if I'm reading it correctly.

Reply:
Hmm... true. I did a bit of refactoring so we avoid calling it all. It's a bit annoying to try to avoid triggering `unused_*` lints.